Garmin.Connect/Models/GarminWorkoutSegment.cs

```diff
@@ -0,0 +1,168 @@
+using System.Text.Json.Serialization;
+
+namespace Garmin.Connect.Models;
+
+public record GarminWorkoutSegment
+{
+    [JsonPropertyName("segmentOrder")]
+    public long SegmentOrder { get; init; }
+
+    [JsonPropertyName("sportType")]
+    public GarminSportType SportType { get; init; }
+
+    [JsonPropertyName("workoutSteps")]
+    public GarminWorkoutStep[] WorkoutSteps { get; init; }
+}
+
+public record GarminWorkoutStep
+{
+    [JsonPropertyName("type")]
+    public string Type { get; init; }
```

Comment:
Use operator `with` (you can see in [UpdateWorkout](https://github.com/sealbro/dotnet.garmin.connect/blob/workouts_v1/Garmin.Connect.Tests/Integrations/WorkoutTests.cs#L48C3-L70) test) because I follow the immutable approach in this library.